### DISCUSSION TITLE
GitHub action unable to set condition on job based on environment variable

### DISCUSSION THREAD

**Discussion by SPodjasek** [community/community#35415] [Actions] [answered]
I have a globally declared environment variable in my workflow, and I wanted to condition job execution based on that value - but it throws a syntax error:

```
The workflow is not valid. .github/workflows/publish-container.yml (Line: 131, Col: 9): 
  Unrecognized named-value: 'env'. Located at position 1 within expression: env.PUBLISH_AND_NOTIFY != 'true'
```

It works as expected when I move this condition and copy it on the job level (using the same expression). However, looking at the documentation, I couldn't find any information that using `env` context on the job level was impossible - is this a bug?

**Comment by airtower-luna** [comment, marked as answer]
It's documented in the [`env` context description](https://docs.github.com/en/actions/learn-github-actions/contexts#env-context):

> You can access this context from any step in a job.

A job-level `if` isn't part of a step, so the behavior is expected.

**Reply by SPodjasek** [reply]
How could I not notice it :see_no_evil: - anyway thanks a lot.

**Comment by bormm** [comment]
It's so sad that these simple things are impossible to do in GitHub.
